pkg/common/client/config.go

```diff
@@ -38,52 +39,202 @@ func MergeSaramaSettings(config *sarama.Config, saramaSettingsYamlString string)
+				saramaConf.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &XDGSCRAMClient{HashGeneratorFcn: SHA256} }
+				saramaConf.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA256
+			}
+
+			if kafkaAuthCfg.SASL.SaslType == sarama.SASLTypeSCRAMSHA512 {
+				saramaConf.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &XDGSCRAMClient{HashGeneratorFcn: SHA512} }
+				saramaConf.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA512
+			}
+			saramaConf.Net.SASL.User = kafkaAuthCfg.SASL.User
+			saramaConf.Net.SASL.Password = kafkaAuthCfg.SASL.Password
+		}
 	}
+	return nil
+}
 
-	// Return Success
-	return config, nil
+// ConfigEqual is a convenience function to determine if two given sarama.Config structs are identical aside
+// from unserializable fields (e.g. function pointers).  To ignore parts of the sarama.Config struct, pass
+// them in as the "ignore" parameter.
+func ConfigEqual(config1, config2 *sarama.Config, ignore ...interface{}) bool {
+	// If some of the types in the sarama.Config struct are not ignored, these kinds of errors will appear:
+	// panic: cannot handle unexported field at {*sarama.Config}.Consumer.Group.Rebalance.Strategy.(*sarama.balanceStrategy).name
+
+	// Note that using the types directly from config1 is convenient (it allows us to call IgnoreTypes instead of the
+	// more complicated IgnoreInterfaces), but it will fail if, for example, config1.Consumer.Group.Rebalance is nil
+
+	// However, the sarama.NewConfig() function sets all of these values to a non-nil default, so the risk
+	// is minimal and should be caught by one of the several unit tests for this function if the sarama vendor
+	// code is updated and these defaults become something invalid at that time)
+
+	ignoreTypeList := append([]interface{}{
+		config1.Consumer.Group.Rebalance.Strategy,
+		config1.MetricRegistry,
+		config1.Producer.Partitioner},
+		ignore...)
+	ignoredTypes := cmpopts.IgnoreTypes(ignoreTypeList...)
+
+	// If some interfaces are not included in the "IgnoreUnexported" list, these kinds of errors will appear:
+	// panic: cannot handle unexported field at {*sarama.Config}.Net.TLS.Config.mutex: "crypto/tls".Config
+
+	// Note that x509.CertPool and tls.Config are created here explicitly because config1/config2 may not
+	// have those fields, and results in a nil pointer panic if used in the IgnoreUnexported list indirectly
+	// like config1.Version is (Version is required to be present in a sarama.Config struct).
+
+	ignoredUnexported := cmpopts.IgnoreUnexported(config1.Version, x509.CertPool{}, tls.Config{})
+
+	// Compare the two sarama config structs, ignoring types and unexported fields as specified
+	return cmp.Equal(config1, config2, ignoredTypes, ignoredUnexported)
 }
 
-// Utility Function For Configuring Common Settings For Admin/Producer/Consumer
-func UpdateSaramaConfig(config *sarama.Config, clientId string, username string, password string) {
+type KafkaAuthConfig struct {
```

Comment:
It would be great to keep the type declaration at the top of the file

Reply:
Sure